### PR TITLE
chore: added intercom consent analytics event

### PR DESCRIPTION
**Pull request by ankitsrivas14**
Added intercom consent analytics event
>
>
#### PR fixes following issue(s)
Fixes 
https://github.com/appsmithorg/cloud-services/issues/1271

#### Media
image

>
#### Type of change
- New feature (non-breaking change which adds functionality)
- Chore (housekeeping or task changes that don't impact user perception)
>
>
>
- [x] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
  - Introduced a new analytics event "SUPPORT_REQUEST_INITIATED" to track support interactions.
  
- **Enhancements**
  - Help button now includes functionality to securely send user data to Intercom and log support interactions for improved customer support experience.

### CODE REVIEW

File: app/client/src/pages/Editor/HelpButton.tsx [hunk 1]
```diff
@@ -114,6 +116,20 @@ export function IntercomConsent({
...
+    if (user?.enableTelemetry) {
+      AnalyticsUtil.identifyUser(user, true);
+      AnalyticsUtil.logEvent("SUPPORT_REQUEST_INITIATED", {
+        email,
+        userId: sha256(username || ""),
+        appsmithVersion: `Appsmith ${!cloudHosting ? appVersion.edition : ""} ${
+          appVersion.id
+        }`,
+        instanceId,
+        id: sha256(username || ""),
```
Comment (ankitakinger) [inline app/client/src/pages/Editor/HelpButton.tsx:129]: in the logEvent function, we already have the keys for `instanceId`, `version`, `userId`, `email`. Could you confirm with the team if we need to add this separately again?

Also, why 2 keys `id` and `userId` with same information is required?

File: app/client/src/utils/AnalyticsUtil.tsx [hunk 2]
```diff
@@ -205,6 +205,14 @@ class AnalyticsUtil {
         const userProperties = {
           userId: AnalyticsUtil.cachedAnonymoustId,
           source: "ce",
+          ...(sendAdditionalData
+            ? {
+                id: AnalyticsUtil.cachedAnonymoustId,
```
Comment (ankitakinger) [inline app/client/src/utils/AnalyticsUtil.tsx:210]: again, why 2 keys id and userId?

File: app/client/src/pages/Editor/HelpButton.tsx (hunk 1 above)
Comment (ankitsrivas14) [reply app/client/src/pages/Editor/HelpButton.tsx]: Hey, I've removed the duplicate keys and kept only `email` as for self-hosted today we do not send email info.

File: app/client/src/utils/AnalyticsUtil.tsx (hunk 2 above)
Comment (ankitsrivas14) [reply app/client/src/utils/AnalyticsUtil.tsx:210]: This is because of historical reasons we have to maintain two keys now.

Comment (ankitakinger) [reply app/client/src/utils/AnalyticsUtil.tsx:210]: I hope you have confirmed this with the team.

**Review by ankitakinger — APPROVED** [review]
LGTM

**Pull request merged**